### PR TITLE
Added GameBoardBlock

### PR DESCRIPTION
**Pull request by kavanparam**
- Created new GameBoardBlock component to contain state for each grid item in GameBoard component
- Added state & UI to each block such that selected blocks are highlighted

image

### CODE REVIEW

File: components/math/multiplicationConnect/GameBoard.tsx [hunk 1]
```diff
@@ -1,37 +1,23 @@
...
+import GameBoardBlock from './GameBoardBlock';
 
-const createGrid = () => {
-    let arr = [];
-    let grid = [];
 
-    for(let i=4; i<25; i++){
-        (i%2===0) ? arr.push(i) : "";
-    }
+const GameBoard = ({grid}) => {
```
Comment (vithushan19) [inline components/math/multiplicationConnect/GameBoard.tsx:5]: create a type for this prop object

`type GameBoardProps = {...}`

File: components/math/multiplicationConnect/GameBoardBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, {useState} from 'react'
+
+
+const blockClick = (block) => {
```
Comment (vithushan19) [inline components/math/multiplicationConnect/GameBoardBlock.tsx:4]: can you add a type annotation to this `block` input parameter

File: components/math/multiplicationConnect/GameBoardBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    (block.isSelected===false) ? block.isSelected=true : block.isSelected=false;
+    console.log(block);
```
Comment (vithushan19) [inline components/math/multiplicationConnect/GameBoardBlock.tsx:6]: don't merge console logs into main

File: components/math/multiplicationConnect/GameBoardBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+const GameBoardBlock = ({ blockData, onClick }) => {
+  const [block, setBlock] = useState(blockData);
+
+  return (
+    <div
+      onClick={()=>setBlock(blockClick(block))}
+      className={`flex justify-center items-center h-full w-full cursor-pointer rounded-full duration-300 
+            ${block.isSelected===false ? "hover:bg-[#F20000]/40 hover:animate-pulse" : "bg-[#F20000]/40"}`}
```
Comment (vithushan19) [inline components/math/multiplicationConnect/GameBoardBlock.tsx:17]: nice!

File: pages/studentPortal/labs/multiplication-connect/Index.tsx [hunk 5]
```diff
@@ -1,16 +1,42 @@
-import React from 'react'
+import React, { useState } from 'react'
 import DiceSection from '../../../../components/math/multiplicationConnect/DiceSection';
 import GameBoard from '../../../../components/math/multiplicationConnect/GameBoard';
 import PlayerSection from '../../../../components/math/multiplicationConnect/PlayerSection';
+import { getRandomItemFromArray } from '../../../api/random';
+
+const calculateWinner = () => {
```
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication-connect/Index.tsx:7]: why do you have an empty function? should we remove it?

**Review by vithushan19 — APPROVED** [review]
looks good to me, left some comments about adding some types, but looks like you got the state logic to work

File: pages/studentPortal/labs/multiplication-connect/Index.tsx (hunk 5 above)
Comment (kavanparam) [reply pages/studentPortal/labs/multiplication-connect/Index.tsx:7]: Yeah I didn't intend for this to be in this PR. Should've created a new commit.

**Pull request merged**